### DISCUSSION TITLE
Fluids

### DISCUSSION THREAD

**Discussion by MrGeoTech** [Minestom/Minestom#794] [Help] [answered]
How can I make water and lava flow? I know that fluids and blocks are two different things but there doesn't seem to be a way to add fluids to Instances. setBlock() doesn't work because it's not a block and there is no FALLING_WATER block.

**Comment by EpicPlayerA10** [comment, marked as answer]
You can use `Block.WATER` and add properties. You can read more about properties in https://minecraft.fandom.com/wiki/Water#Block_states

**Reply by MrGeoTech** [reply]
I was looking at doing this. I just wanted to see if Fluids were supposed to be used for this. Which then raises the question, what are they used for and is there even a point in keeping them?

**Reply by EpicPlayerA10** [reply]
In minestom `Fluids` class is useless. You can't use them in any api method.

**Reply by MrGeoTech** [reply]
K. Good to know. Thanks

**Reply by BomBardyGamer** [reply]
`Fluids` is a relic that exists because vanilla has a separation between blocks and fluids (as in two completely separate, but very similar systems), and Minestom needed a way to represent that easily for data purposes without writing a full fluid system like vanilla.
